packages/core-plugin-api/src/translation/useTranslationRef.ts

```diff
@@ -14,36 +14,61 @@
  * limitations under the License.
  */
 
-import { useTranslation } from 'react-i18next';
+import { useEffect, useState } from 'react';
 
 import { useApi } from '../apis';
-import { appTranslationApiRef } from '../apis/alpha';
-import { toInternalTranslationRef, TranslationRef } from './TranslationRef';
-
-/** @alpha */
-export interface TranslationOptions {
-  /* no options supported for now */
-}
+import {
+  translationApiRef,
+  TranslationFunction,
+  TranslationSnapshot,
+} from '../apis/alpha';
+import { TranslationRef } from './TranslationRef';
 
 /** @alpha */
 export const useTranslationRef = <
   TMessages extends { [key in string]: string },
 >(
   translationRef: TranslationRef<string, TMessages>,
-) => {
-  const translationApi = useApi(appTranslationApiRef);
+): { t: TranslationFunction<TMessages> } => {
+  const translationApi = useApi(translationApiRef);
+
+  const [error, setError] = useState<Error>();
+  const [snapshot, setSnapshot] = useState<TranslationSnapshot<TMessages>>(() =>
+    translationApi.getTranslation(translationRef),
+  );
+  const [observable] = useState(() =>
+    translationApi.translation$(translationRef),
+  );
+
+  useEffect(() => {
+    const subscription = observable.subscribe({
+      next(next) {
+        if (next.ready) {
+          setSnapshot(next);
+        }
+      },
+      error: setError,
+    });
 
-  const internalRef = toInternalTranslationRef(translationRef);
-  translationApi.addResource(translationRef);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [observable]);
 
-  const { t } = useTranslation(internalRef.id, {
-    useSuspense: process.env.NODE_ENV !== 'test',
-  });
+  if (error) {
+    throw error;
+  }
 
-  const defaultMessages = internalRef.getDefaultMessages();
+  if (!snapshot.ready) {
+    throw new Promise<void>(resolve => {
```

Comment:
Ah yeah probably, well add a test

Reply:
Yep, did indeed need to improve the error handling. Figured for now that logging failed resource loads is the way to go. If that causes trouble we could also extend the snapshots to have an error property, but not really sure it would be desirable to crash to error boundary anyway.

Reply:
Probably isn't. May be good to stay on a "latest known good" state (even reverting to default translations if needed) and just signal to errorApi that something went wrong.

Reply:
👍 added errorApi post + some more fixes